sdks/java/core/src/main/java/org/apache/beam/sdk/io/BlockBasedSource.java

```diff
@@ -207,27 +207,29 @@ protected final boolean readNextRecord() throws IOException {
 
     @Override
     public Double getFractionConsumed() {
-      if (getCurrentSource().getEndOffset() == Long.MAX_VALUE) {
-        return null;
-      }
-      Block<T> currentBlock = getCurrentBlock();
-      if (currentBlock == null) {
-        // There is no current block (i.e., the read has not yet begun).
+      if (!isStarted()) {
         return 0.0;
       }
+      if (isDone()) {
+        return 1.0;
+      }
+      FileBasedSource<T> source = getCurrentSource();
+      if (source.getEndOffset() == Long.MAX_VALUE) {
```

Comment:
Why does this return null? Does it indicate that there is an unlimited amount of data? Or just that we don't know how much data is left, therefore we can't estimate what fraction we've consumed?